src/components/shared/thread/ActivityGraph.js

```diff
@@ -0,0 +1,322 @@
+    const canvas = this._canvas;
+    if (!canvas || !activityGraphFills) {
+      return;
+    }
+    // Re-measure the canvas and get the coordinates and time for the click.
+    const { rangeStart, rangeEnd } = this.props;
+    const rect = canvas.getBoundingClientRect();
+    const x = e.pageX - rect.left;
+    const y = e.pageY - rect.top;
+    const time = rangeStart + x / rect.width * (rangeEnd - rangeStart);
+
+    const sample = activityGraphFills.getSampleAtClick(x, y, time);
+    if (sample !== null) {
+      this.props.onSampleClick(sample);
+    }
+  };
+
+  render() {
+    this._renderCanvas();
+    return (
+      <div className={this.props.className}>
+        <canvas
+          className={classNames(
+            `${this.props.className}Canvas`,
+            'threadActivityGraphCanvas'
+          )}
+          ref={this._takeCanvasRef}
+          onMouseUp={this._onMouseUp}
+        />
+      </div>
+    );
+  }
+}
+
+export default ThreadActivityGraph;
+
+/**
+ * Filtered out samples use a diagonal stripe pattern, create that here.
+ */
+function _createDiagonalStripePattern(
+  chartCtx: CanvasRenderingContext2D,
+  color: string
+): CanvasPattern {
+  // Create a second canvas, draw to it in order to create a pattern. This canvas
+  // and context will be discarded after the pattern is created.
+  const patternCanvas = document.createElement('canvas');
+  const dpr = Math.round(window.devicePixelRatio);
+  patternCanvas.width = 4 * dpr;
+  patternCanvas.height = 4 * dpr;
+  const patternContext = patternCanvas.getContext('2d');
+  patternContext.scale(dpr, dpr);
+
+  const linear = patternContext.createLinearGradient(0, 0, 4, 4);
+  linear.addColorStop(0, color);
+  linear.addColorStop(0.25, color);
+  linear.addColorStop(0.25, 'transparent');
+  linear.addColorStop(0.5, 'transparent');
+  linear.addColorStop(0.5, color);
+  linear.addColorStop(0.75, color);
+  linear.addColorStop(0.75, 'transparent');
+  linear.addColorStop(1, 'transparent');
+  patternContext.fillStyle = linear;
+  patternContext.fillRect(0, 0, 4, 4);
+
+  return chartCtx.createPattern(patternCanvas, 'repeat');
+}
+
+/**
+ * Map a color name, which comes from Gecko, into a CSS style color. These colors cannot
+ * be changed without considering the values coming from Gecko, and from old profiles
+ * that already have their category colors saved into the profile.
+ *
+ * Category color names come from:
+ * https://searchfox.org/mozilla-central/rev/0b8ed772d24605d7cb44c1af6d59e4ca023bd5f5/tools/profiler/core/platform.cpp#1593-1627
+ */
+function _mapColorNameToStyles(colorName: string) {
+  switch (colorName) {
+    case 'transparent':
+      return {
+        selectedFillStyle: 'transparent',
+        unselectedFillStyle: 'transparent',
+        gravity: 0,
+      };
+    case 'purple':
+      return {
+        selectedFillStyle: photonColors.PURPLE_70,
+        unselectedFillStyle: photonColors.PURPLE_70 + '60',
```

Comment:
We should probably add a comment somewhere around here about the fact that we're assuming the photonColors to have the form `#RRGGBB`, so that adding `60` at the end will change it to the form `#RRGGBBAA` and make it more transparent.